src/librustdoc/visit_ast.rs

```diff
@@ -31,6 +31,19 @@ fn def_id_to_path(tcx: TyCtxt<'_>, did: DefId) -> Vec<String> {
     std::iter::once(crate_name).chain(relative).collect()
 }
 
+pub(crate) fn inherits_doc_hidden(cx: &core::DocContext<'_>, mut node: hir::HirId) -> bool {
+    while let Some(id) = cx.tcx.hir().get_enclosing_scope(node) {
+        node = id;
+        if cx.tcx.hir().attrs(node).lists(sym::doc).has_word(sym::hidden) {
+            return true;
+        }
+        if node == hir::CRATE_HIR_ID {
+            break;
+        }
```

Comment:
Is this break necessary? It seems like the enclosing scope for the crate will be `None`, no?

Reply:
From what I understand from the source code of `get_enclosing_scope`, it'll return itself in such a case, creating an infinite loop, which is why we need this `break`.

Reply:
That's not right, `ParentHirIterator` returns `None` for the crate root: https://github.com/rust-lang/rust/blob/5fa22fe6f821ac3801d05f624b123dda25fde32c/compiler/rustc_middle/src/hir/map/mod.rs#L119-L120

So I would remove this check.